pkg/scheduler/objects/queue.go

```diff
@@ -669,6 +669,52 @@ func (sq *Queue) GetPartitionQueueDAOInfo() dao.PartitionQueueDAOInfo {
 	return queueInfo
 }
 
+// GetSingleQueueDAOInfo returns the single queue as an object for a REST call.
+func (sq *Queue) GetSingleQueueDAOInfo() dao.PartitionQueueDAOInfo {
+	queueInfo := dao.PartitionQueueDAOInfo{}
+	childes := sq.GetCopyOfChildren()
+	queueInfo.ChildrenNames = make([]string, 0, len(childes))
+	for _, child := range childes {
+		queueInfo.ChildrenNames = append(queueInfo.ChildrenNames, child.Name)
+	}
+	// we have held the read lock so following method should not take lock again.
```

Comment:
those code is almost equal to `GetPartitionQueueDAOInfo`. Could you do a bit refactor for them?